presto-testing/src/main/java/io/prestosql/testing/datatype/DataTypeTest.java

```diff
@@ -55,27 +72,66 @@ public void execute(QueryRunner prestoExecutor, Session session, DataSetup dataS
         List<Object> expectedResults = inputs.stream().map(Input::toPrestoQueryResult).collect(toList());
         try (TestTable testTable = dataSetup.setupTestTable(unmodifiableList(inputs))) {
             MaterializedResult materializedRows = prestoExecutor.execute(session, "SELECT * from " + testTable.getName());
-            assertEquals(materializedRows.getTypes(), expectedTypes);
-            List<Object> actualResults = getOnlyElement(materializedRows).getFields();
-            assertEquals(actualResults.size(), expectedResults.size(), "lists don't have the same size");
-            for (int i = 0; i < expectedResults.size(); i++) {
-                assertEquals(actualResults.get(i), expectedResults.get(i), "Element " + i);
+            checkResults(expectedTypes, expectedResults, materializedRows);
+            if (runSelectWithWhere) {
+                String prestoQuery = buildPrestoQueryWithWhereClauses(testTable);
+                MaterializedResult filteredRows = prestoExecutor.execute(session, prestoQuery);
+                checkResults(expectedTypes, expectedResults, filteredRows);
+            }
+        }
+    }
+
+    private String buildPrestoQueryWithWhereClauses(TestTable testTable)
+    {
+        List<String> predicates = new ArrayList<>();
+        for (int i = 0; i < inputs.size(); i++) {
+            Input<?> input = inputs.get(i);
+            if (input.isUseInWhereClause()) {
```

Comment:
why do we need this?

Reply:
We may want to disable some values (NaN, Inf, etc.) from the WHERE clause, since I'm not sure such values would behave correctly in the `IS NOT DISTINCT FROM value` comparison.